### ISSUE TITLE
improve sys.log indices

### ISSUE THREAD

**Issue by m-rau** [plan-net/core4#116]
To improve performance of ``chist`` (core history) the following options should have better ``sys.log`` indices:

```
--project=PROJECT
--hostname=HOSTNAME
--username=USERNAME
--qual_name=QUAL_NAME
--identifier=ID
```

**Comment by m-rau** [comment]
requires indices on
* hostname
* identifier
* username
* qual_name

**Comment by m-rau** [comment]
has been improved. Requires dropping the collection sys.log before this change can have an effect.

**Comment by m-rau** [comment]
chist/sys.log still too slow.
Needs further improvements as the indices are obviously not used.

**Comment by m-rau** [comment]
The issue is with sys.log being a capped collection. Querying is performant if the sort order is $natural. In all other cases it uses a full scan.

### suggested solution
convert capped collection into a regular collection

**Comment by basteiz** [comment]
Cause is:

https://docs.mongodb.com/v3.6/reference/operator/query/regex/

see last paragraph case insensitive

`Index Use

For case sensitive regular expression queries, if an index exists for the field, then MongoDB matches the regular expression against the values in the index, which can be faster than a collection scan. Further optimization can occur if the regular expression is a “prefix expression”, which means that all potential matches start with the same string. This allows MongoDB to construct a “range” from that prefix and only match against those values from the index that fall within that range.

A regular expression is a “prefix expression” if it starts with a caret (^) or a left anchor (\A), followed by a string of simple symbols. For example, the regex /^abc.*/ will be optimized by matching only against the values from the index that start with abc.

Additionally, while /^a/, /^a.*/, and /^a.*$/ match equivalent strings, they have different performance characteristics. All of these expressions use an index if an appropriate index exists; however, /^a.*/, and /^a.*$/ are slower. /^a/ can stop scanning after matching the prefix.

**Case insensitive regular expression queries generally cannot use indexes effectively. The $regex implementation is not collation-aware and is unable to utilize case-insensitive indexes.**`

example query:

`command core4.sys.log command: find { find: "sys.log", filter: { $and: [ { created: { $gte: new Date(1578926814461) } }, **{ identifier: /^.*5e1c90e4ec6c92880893ca09$/iu }** ] }, sort: { _id: -1 }, lsid: { id: UUID("bf1297a4-03d2-4530-b184-61e5e667156e") }, $clusterTime: { clusterTime: Timestamp(1578930414, 2), signature: { hash: BinData(0, 974523E47035F9D709A9F2A8D4B7540B743E02B7), keyId: 6717527853958692873 } }, $db: "core4", $readPreference: { mode: "primary" } } planSummary: IXSCAN { _id: 1 } keysExamined:12221794 docsExamined:12221794 cursorExhausted:1 numYields:95484 nreturned:3 reslen:999 locks:{ Global: { acquireCount: { r: 190970 } }, Database: { acquireCount: { r: 95485 } }, Collection: { acquireCount: { r: 95485 } } } protocol:op_msg 46309ms`

no index is used, neither the created, nor the identifier!

see index:

`{
    "created" : -1,
    "identifier" : -1
}`

Solution:

implementation without case insensitive

**Comment by m-rau** [comment]
I guess your suggestion is not the complete story. Capped collections rely heavily on the natural sort index. This cannot be resolved with chist search requirements.
Therefore, the collection is to be uncapped.

**Comment by m-rau** [comment]
users still report chist is very slow. Further analysis shows that the mongodb cursor is not really used.
Furthermore --explain option introduced with chist to analyse index usage.
This has been fixed with a hotfix into master. Rational is to rollout and test quickly.

**Comment by m-rau** [comment]
please retest

**Comment by m-rau** [comment]
fixed by 1aa407a8c1aea343fa11f06d91102f33a7885b8c
